src/search/merge_and_shrink/merge_and_shrink_algorithm.cc

```diff
@@ -55,9 +56,61 @@ MergeAndShrinkAlgorithm::MergeAndShrinkAlgorithm(
       log(utils::get_log_for_verbosity(verbosity)),
       main_loop_max_time(main_loop_max_time),
       starting_peak_memory(0) {
-    assert(max_states_before_merge > 0);
-    assert(max_states >= max_states_before_merge);
-    assert(shrink_threshold_before_merge <= max_states_before_merge);
+    handle_shrink_limit_defaults();
+
+    assert(this->max_states_before_merge > 0);
```

Comment:
Do we need `this` here? It looks like a different setting than before because there are no templates involved here.

Reply:
Without `this` it would mean the parameter (which is not changed by `handle_shrink_limit_defaults`), with `this` it means the field (which happens to have the same name).

Reply:
I see. This is a bit unintuitive. Maybe add a short comment explaining this.

Reply:
We could also have a `_` prefix for the parameters.

Reply:
We decided against this, the last time we discussed this issue. I'd go with a comment even if we changed the name.

Reply:
I added a comment